fullstop-timemachine/src/main/java/org/zalando/stups/fullstop/swagger/api/FullstopApi.java

```diff
@@ -66,6 +66,49 @@
     @Autowired
     private TeamOperations teamOperations;
 
+    private static Violation mapToDto(ViolationEntity entity) {
+
+        if (entity == null) {
+            return null;
+        }
+
+        Violation violation = new Violation();
+
+        violation.setId(entity.getId());
+        violation.setVersion(entity.getVersion());
+
+        violation.setCreated(entity.getCreated());
+        violation.setCreatedBy(entity.getCreatedBy());
+        violation.setLastModified(entity.getLastModified());
+        violation.setLastModifiedBy(entity.getLastModifiedBy());
+
+        violation.setAccountId(entity.getAccountId());
+        violation.setEventId(entity.getEventId());
+        violation.setMessage(entity.getMessage());
+        violation.setRegion(entity.getRegion());
+        violation.setComment(entity.getComment());
+        violation.setViolationObject(entity.getViolationObject());
+        return violation;
+    }
+
+    @ApiOperation(
+            value = "violations", notes = "Get one violation", response = Violation.class
+    )
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "Violation") })
+    @PreAuthorize("#oauth2.hasScope('uid')")
+    @RequestMapping(value = "/violations/{id}", method = RequestMethod.GET)
+    public Violation getViolation(
+            @ApiParam(value = "Violation id")
+            @PathVariable(value = "id")
+            final Long id,
+            @AuthenticationPrincipal(errorOnInvalidType = true) final String uid) throws NotFoundException {
```

Comment:
not needed right now. maybe in the future when we add also the "get violation for only my team" functionality.